src/Uno.UI/UI/Xaml/Documents/LanguageTagCache.skia.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using Windows.Globalization;
+
+#nullable enable
+
+namespace Windows.UI.Xaml.Documents
+{
+	internal class LanguageTagCache
+	{
+		[ThreadStatic] private static readonly Dictionary<string, string[]> _languageTagsCache = new();
+
+		public static string[] GetLanguageTag(TextElement element)
+		{
+			var key = element.Language
+				?? ApplicationLanguages.PrimaryLanguageOverride
+				?? string.Empty;
+			if(!_languageTagsCache.TryGetValue(key,out var tag))
+			{
+				var cultureInfo = string.IsNullOrEmpty(key)
```

Comment:
`key` can't be null at this point

```suggestion
				var cultureInfo = key.Length == 0
```